app/views/related_works/index.html.erb

```diff
@@ -132,23 +134,17 @@
       <% unless related_work.translation? %>
         <% if related_work.parent && related_work.work %>
           <dt class="parent">
-            <% if related_work.parent.is_a?(ExternalWork) %>
-               <%= link_to related_work.parent.title, related_work.parent %> by <%= byline(related_work.parent) %>
-            <% elsif related_work.parent.unrevealed? || related_work.parent.anonymous? %>
-              <%= ts("A work in an unrevealed or anonymous challenge") %>
+            <% if related_work.parent.is_a?(Work) && related_work.parent.unrevealed? %>
+              <%= ts("A work in an unrevealed collection") %>
             <% else %>
               <%= link_to related_work.parent.title, related_work.parent %> <%= ts("by") %> <%= byline(related_work.parent) %>
             <% end %>
           </dt>
- 		      <dd class="child">
- 		        <%= ts("inspired") %>
-            <% if related_work.work.unrevealed? || related_work.work.anonymous? %>
-              <%= ts("A work in an unrevealed or anonymous challenge") %>
-            <% else %>
-              <%= link_to related_work.work.title, related_work.work %>
-            <% end %>
+          <dd class="child">
+            <%= ts("inspired") %>
+            <%= link_to related_work.work.title, related_work.work %>
             <% if current_user == @user %>
-              <span class="actions" role="button"><%= link_to ts("Remove"), {:controller => :related_works, :action => :destroy, :id => related_work.id},  data: {confirm: ts('Are you sure?')}, :method => :delete %></span>
+              <span class="actions" role="button"><%= button_to ts("Remove"), related_work_path(related_work), data: { confirm: ts("Are you sure?") }, method: :delete %></span>
```

Comment:
Could you remove the `role="button"` part? I'm not sure it ever should have been used here, but it definitely is unnecessary now that we have an actual button.